CMakeLists.txt

```diff
@@ -13,106 +13,89 @@ include(cmake/util/Util.cmake)
 include(cmake/util/MshadowUtil.cmake)
 include(cmake/util/FindCUDA.cmake)
 
-# TODO(#5475): Clean up the old flags after CI and regression framework adopt to the new setup.
-if (NOT DEFINED BUILD_TYPE)
-  dgl_option(USE_CUDA "Build with CUDA" OFF)
-  dgl_option(USE_OPENMP "Build with OpenMP" ON)
-  dgl_option(USE_LIBXSMM "Build with LIBXSMM library optimization" ON)
-  dgl_option(BUILD_CPP_TEST "Build cpp unittest executables" OFF)
-  dgl_option(LIBCXX_ENABLE_PARALLEL_ALGORITHMS "Enable the parallel algorithms library. This requires the PSTL to be available." OFF)
-  dgl_option(USE_S3 "Build with S3 support" OFF)
-  dgl_option(USE_HDFS "Build with HDFS support" OFF) # Set env HADOOP_HDFS_HOME if needed
-  dgl_option(REBUILD_LIBXSMM "Clean LIBXSMM build cache at every build" OFF) # Set env HADOOP_HDFS_HOME if needed
-  dgl_option(USE_EPOLL "Build with epoll for socket communicator" ON)
-  dgl_option(BUILD_TORCH "Build the PyTorch plugin" OFF)
-  dgl_option(BUILD_SPARSE "Build DGL sparse library" ON)
-  dgl_option(BUILD_GRAPHBOLT "Build Graphbolt library" OFF)
-  dgl_option(TORCH_PYTHON_INTERPS "Python interpreter used to build tensoradapter and DGL sparse library" python3)
-else()
-  # Options for building DGL.
-  # NOTE: do not modify this file to change option values.
-  # Use bash script/build_dgl.sh -e '-DOPTION=VALUE' through command-line.
-  dgl_option(
-      BUILD_TYPE
-      "Type of the build: dev, test or release"
-      "dev"
-  )
-  message(STATUS "Build for ${BUILD_TYPE}")
+# Options for building DGL.
+# NOTE: do not modify this file to change option values.
+# Use bash script/build_dgl.sh -e '-DOPTION=VALUE' through command-line.
+dgl_option(
+    BUILD_TYPE
```

Comment:
we'd better make sure all builders have already turned to this new way: `BUILD_TYPE`? Or at least print a warning message if not defined.

Reply:
no, we just need to make sure that our release build pipeline and test pipeline are using proper BUILD_TYPE.